blockchain/tx_list.go

```diff
@@ -281,14 +312,24 @@ func (l *txList) Add(tx *types.Transaction, priceBump uint64) (bool, *types.Tran
 	// If there's an older better transaction, abort
 	old := l.txs.Get(tx.Nonce())
 	if old != nil {
-		if tx.Type().IsCancelTransaction() {
-			logger.Trace("New tx is a cancel transaction. replace it!", "old", old.String(), "new", tx.String())
-		} else {
+		// TODO : Need to KIP-71 hardfork.
+		//if tx.Type().IsCancelTransaction() {
+		//	logger.Trace("New tx is a cancel transaction. replace it!", "old", old.String(), "new", tx.String())
+		//} else {
+		//	logger.Trace("already nonce exist", "nonce", tx.Nonce(), "with gasprice", old.GasPrice(), "priceBump", priceBump, "new tx.gasprice", tx.GasPrice())
+		//	return false, nil
+		//}
+
+		// If gas price of older is bigger than newer, abort.
+		if old.GasPrice().Cmp(tx.GasPrice()) >= 0 {
 			logger.Trace("already nonce exist", "nonce", tx.Nonce(), "with gasprice", old.GasPrice(), "priceBump", priceBump, "new tx.gasprice", tx.GasPrice())
 			return false, nil
```

Comment:
@sirano11 @kjhman21 I want to confirm the policy of the cancel transaction. Which option are you thinking?
1. Cancel transaction can cancel a transaction having the same nonce regardless of gas price 
2. Cancel transaction is handled as other transaction types, so it can cancel a transaction only when it has more gas price than the predecessor.

Reply:
@aidan-kwon Please remove a duplicated review.